### PR TITLE
Fix edgecases for lineinfile module

### PR DESCRIPTION
**Pull request by sairon**
Edgecases where the file contains only a single line and no before/after regexp was given were not handled correctly. Also the Find method should accept empty string for searching from the beginning of the file.

Moreover, it's useful if the file can be created by Present (it's also what Ansible does), so handle this case too.

## Summary by CodeRabbit

- **Refactor**
	- Improved file handling and line processing logic for enhanced efficiency and clarity.
- **Tests**
	- Expanded test coverage to verify behavior across various scenarios and edge cases.

### CODE REVIEW

File: utils/lineinfile/lineinfile.go [hunk 1]
```diff
@@ -47,16 +47,23 @@ func NewAbsentParams() Params {
 }
 
 func (l LineInFile) Present(params Params) error {
```
Comment (agners) [inline utils/lineinfile/lineinfile.go:49]: Some docs would be nice what this function is doing (I guess check if a particular parameter is present).

File: utils/lineinfile/lineinfile.go [hunk 2]
```diff
@@ -47,16 +47,23 @@ func NewAbsentParams() Params {
...
+	} else if err != nil {
 		return err
 	}
 
-	content, err := os.ReadFile(l.FilePath)
-	if err != nil {
-		return err
-	}
+	var lines []string
+	if !createFile {
```
Comment (agners) [inline utils/lineinfile/lineinfile.go:59]: It this is only checking if something is present, then creating a file seems a rather surprising side effect :thinking:

File: utils/lineinfile/lineinfile.go (hunk 1 above)
Comment (sairon) [reply utils/lineinfile/lineinfile.go:49]: It was already added in #207 so it's a bit unrelated to this PR but I can add some docs later. As explained in the other PR, it's akin to Ansible's lineinfile module, where "present" ensures that a line in file exists, "absent" ensures it's not.

File: utils/lineinfile/lineinfile.go (hunk 2 above)
Comment (sairon) [reply utils/lineinfile/lineinfile.go:59]: As I said in the other comment, `Present` ensures the file contains a given line, so the file must always exist in that case.

**Review by agners — APPROVED** [review]
